### PR TITLE
fetch (logical) replication slot definitions from the master [BF-1213]

### PR DESCRIPTION
**Pull request by carobme**

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -130,6 +130,12 @@ def db():
         )
         if db.pgver < "13":
             fp.write("wal_keep_segments = 100\n")
+
+        if LooseVersion(db.pgver) < LooseVersion("9.4"):
+            fp.write("wal_level = archive\n")
+        else:
+            fp.write("wal_level = logical\n")
```
Comment (carobme) [inline test/conftest.py:137]: `logical` was introduced in 9.4

File: pglookout/cluster_monitor.py [hunk 2]
```diff
@@ -201,6 +224,10 @@ def _query_cluster_member_state(self, instance, db_conn):
                 master_position = c.fetchone()
                 maybe_standby_result["pg_last_xlog_replay_location"] = master_position["pg_last_xlog_replay_location"]
                 f_result = maybe_standby_result
+
+                if db_conn.server_version >= 100000:
```
Comment (carobme) [inline pglookout/cluster_monitor.py:228]: PG < 10 has been EOLed for a while, that's why I didn't bother with anything older here.

File: test/conftest.py (hunk 1 above)
Comment (kmichel-aiven) [reply test/conftest.py:137]: We're not even testing below 9.6, I'd rather not introduce untested/dead code.

File: test/test_cluster_monitor.py [hunk 3]
```diff
@@ -5,12 +5,19 @@
 See LICENSE for details
 """
 
+from contextlib import closing
+from datetime import datetime, timedelta
+from distutils.version import LooseVersion
```
Comment (kmichel-aiven) [inline test/test_cluster_monitor.py:10]: Cf. https://github.com/aiven/pglookout/pull/82 It's probably better to not reintroduce deprecated API usage.

File: test/conftest.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 Copyright (c) 2016 Ohmu Ltd
 See LICENSE for details
 """
+from distutils.version import LooseVersion
```
Comment (kmichel-aiven) [inline test/conftest.py:7]: Cf. https://github.com/aiven/pglookout/pull/82 It's probably better to not reintroduce deprecated API usage.

File: pglookout/cluster_monitor.py [hunk 5]
```diff
@@ -155,6 +155,29 @@ def connect_to_cluster_nodes_and_cleanup_old_nodes(self):
         for instance, connect_string in self.config.get("remote_conns", {}).items():
             self._connect_to_db(instance, dsn=connect_string)
 
+    def _fetch_replication_slot_info(self, instance, cursor):
```
Comment (kmichel-aiven) [inline pglookout/cluster_monitor.py:158]: A list of objects with many keys and values of various types is worth a type hint and a dataclass for the return type :)

**Review by kmichel-aiven — DISMISSED** [review]

File: pglookout/cluster_monitor.py (hunk 5 above)
Comment (carobme) [reply pglookout/cluster_monitor.py:158]: May look a bit weird, as it is the only type hint and dataclass usage in pglookout then, but sure. :-)

File: test/conftest.py (hunk 1 above)
Comment (carobme) [reply test/conftest.py:137]: True, but `PG_VERSIONS` (`conftest.py`) still lists versions back to 9.2, which is why I tried to be compatible with that change. I'm going to remove that conditional, as well as versions older than 9.6 from the mentioned list.

**Review by kmichel-aiven — APPROVED** [review]

**Pull request merged**